src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject.cs

```diff
@@ -27,6 +28,21 @@ internal sealed partial class CPSProject : AbstractProject
             : base(projectTracker, reportExternalErrorCreatorOpt, projectDisplayName, projectFilePath,
                    hierarchy, language, projectGuid, serviceProvider, visualStudioWorkspaceOpt, hostDiagnosticUpdateSourceOpt, commandLineParserServiceOpt)
         {
+            if (binOutputPath != null)
+            {
+                // Ensure that binOutputPath is either null or a rooted path.
+                // CPS might provide such invalid paths during initialization or when project is in unrestored state.
+                if (binOutputPath == String.Empty)
+                {
+                    binOutputPath = null;
+                }
+                else if (!Path.IsPathRooted(binOutputPath))
+                {
+                    // Make it a rooted path.
+                    binOutputPath = Path.Combine(Path.GetTempPath(), binOutputPath);
```

Comment:
What does CPS pass in when the path is unrooted? Is it something like bin\debug? If so, should we make the path be `Path.Combine(Path.GetDirectoryName(projectFilePath), binOutputPath)'?